src/WebJobs.Script/Description/DotNet/FunctionMetadataResolver.cs

```diff
@@ -179,10 +180,25 @@ private ImmutableArray<PortableExecutableReference> GetMetadataFromReferencePath
             }
             else
             {
-                // Treat the reference as a private assembly reference
-                string filePath = Path.Combine(_privateAssembliesPath, reference);
-                if (File.Exists(Path.Combine(filePath)))
+                bool externalReference = false;
+                string basePath = _privateAssembliesPath;
+                
+                // If this is is a relative assembly reference, use the function directory as the base probing path
```

Comment:
If we have a rooted path, we directly add that reference here: https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Description/DotNet/FunctionMetadataResolver.cs#L178